server/src/main/java/org/opensearch/common/settings/ClusterSettings.java

```diff
@@ -709,7 +710,10 @@ public void apply(Settings value, Settings current, Settings previous) {
 
                 // Concurrent segment search settings
                 SearchService.CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING,
-                SearchService.CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_SETTING
+                SearchService.CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_SETTING,
+
+                // Pluggable caches settings
+                StatsHolder.MAX_DIMENSION_VALUES_SETTING
```

Comment:
I don't think we should make this a configurable setting yet (if at all).

Assuming the caches are responsible for defining the set of valid stat dimensions, it should be the responsibility of the authors of a cache implementation to keep the number of dimensions under control.